src/Nethermind/Nethermind.Optimism/Rpc/OptimismEthRpcModule.cs

```diff
@@ -118,7 +117,11 @@ public override async Task<ResultWrapper<Hash256>> eth_sendTransaction(Transacti
 
         if (rpcTx.Nonce is null)
         {
-            tx.Nonce = _accountStateProvider.GetNonce(tx.SenderAddress);
+            // TODO: this kind of defeats the purpose of the refactor - but how do we deal with these situations?
```

Comment:
use _blockFinder.Head and get data from state reader

Reply:
fixed